hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/ReadableNodeStoreImpl.java

```diff
@@ -92,12 +93,18 @@ private Roster constructFromNodesState(@NonNull final ReadableKVState<EntityNumb
         for (final var it = nodesState.keys(); it.hasNext(); ) {
             final var nodeNumber = it.next();
             final var node = requireNonNull(nodesState.get(nodeNumber));
+            final var nodeEndpoints = node.gossipEndpoint();
+            // we want to swap the internal and external node endpoints
+            // so that the external one is at index 0
+            if (nodeEndpoints.size() == 2) {
```

Comment:
Hey @edward-swirldslabs, here do we assume that the `gossipEndpoint` list will always contain exactly 2 endpoints (internal and external in that order). I saw that by definition in DAB HIP it is specified like that but are there other cases to consider?

Reply:
Thanks a lot for the clarification, updated!